src/components/Profile/Activity/ContributionGraph.tsx

```diff
@@ -0,0 +1,205 @@
+// Copyright 2019-2025 @polkassembly/fellowship authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+'use client';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, { useEffect, useState } from 'react';
+import Link from 'next/link';
+import { Tooltip } from '@nextui-org/tooltip';
+import LoadingSpinner from '@/components/Misc/LoadingSpinner';
+import { getUserCommitHistory } from '@/app/api/v1/github/getUserCommitHistory';
+import Image from 'next/image';
+
+function ContributionGraph({ classNames = '', githubUsername, openProfileEdit }: { classNames?: string; githubUsername: string; openProfileEdit: () => void }) {
+	const [contributions, setContributions] = useState<any[]>([]);
+	const [totalContributionsCount, setTotalContributionsCount] = useState(0);
+	const [loading, setLoading] = useState(true);
+	const [error, setError] = useState<string | null>(null);
+
+	const getColor = (count: number) => {
+		if (count === 0) return 'fill-contributionEmpty';
+		if (count < 5) return 'fill-contributionSm';
+		if (count < 10) return 'fill-contributionMd';
+		if (count < 20) return 'fill-contributionLg';
+		return 'fill-contributionXl';
+	};
+
+	useEffect(() => {
+		const fetchContributions = async () => {
+			try {
+				setLoading(true);
+				const { contributions, totalContributionsCount } = await getUserCommitHistory({ username: githubUsername });
+
+				setContributions(contributions);
+				setTotalContributionsCount(totalContributionsCount);
+			} catch (error) {
+				setError((error as Error)?.message);
+			} finally {
+				setLoading(false);
+			}
+		};
+
+		fetchContributions();
+	}, [githubUsername]);
+
+	if (error || !contributions) return <p>Error: {error || 'No contributions found!'}</p>;
+
+	const today = new Date();
+	const lastYear = new Date(today);
+	lastYear.setFullYear(today.getFullYear() - 1);
+	lastYear.setDate(today.getDate() + 1);
+
+	const days = [];
+	for (let day = new Date(lastYear); day <= today; day.setDate(day.getDate() + 1)) {
+		const dateStr = day.toISOString().split('T')[0];
+		const contribution = contributions.find((c) => c.date === dateStr);
+		days.push({
+			date: dateStr,
+			count: contribution ? contribution.count : 0,
+			color: getColor(contribution ? contribution.count : 0)
+		});
+	}
+
+	const weeks = [];
+	for (let i = 0; i < days.length; i += 7) {
+		weeks.push(days.slice(i, i + 7));
+	}
+
+	const months = [];
+	const monthNames = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
+	for (let i = 0; i < 12; i += 1) {
+		const monthIndex = i + 1;
+		months.push({
+			name: monthNames[i],
+			offset: weeks.findIndex((week) => week[0].date.split('-')[1] === monthIndex.toString().padStart(2, '0')) * 12
+		});
+	}
+
+	function formatContributionDate(day: { date: string; count: number; color: string }) {
```

Comment:
a lot of these functions can be made into utils, so that the React component doesn't look too messy